test.cpp

```diff
@@ -238,6 +238,21 @@ TEST(HostnameTest, LowercasesHostname)
     EXPECT_EQ("www.testing.com", Url::Url("http://www.testing.com/FOO").host());
 }
 
+TEST(QueryTest, SanitizesQuery)
+{
+    EXPECT_EQ("a=1&b=2"    , Url::Url("http://foo.com/?a=1&&&&&&b=2"   ).query());
+    EXPECT_EQ("foo=2"      , Url::Url("http://foo.com/????foo=2"       ).query());
+    EXPECT_EQ("foo=2"      , Url::Url("http://foo.com/?foo=2&&&"       ).query());
+}
```

Comment:
I think my preference for the time being is to keep `url-py`'s behavior until this is driving `url-py`, and then we can revisit this case. https://seomoz.atlassian.net/browse/BIG-3870